src/EditorFeatures/Core/Shared/Utilities/ResettableDelay.cs

```diff
@@ -52,8 +55,12 @@ private async Task StartTimerAsync(bool continueOnCapturedContext)
         {
             do
             {
-                // Keep delaying until at least delayInMilliseconds has elapsed since lastSetTime 
-                await Task.Delay(_delayInMilliseconds).ConfigureAwait(continueOnCapturedContext);
+                // Keep delaying until at least delayInMilliseconds has elapsed since lastSetTime
+                if (!await _expeditableDelaySource.Delay(TimeSpan.FromMilliseconds(_delayInMilliseconds), CancellationToken.None).ConfigureAwait(continueOnCapturedContext))
```

Comment:
such a nifty idea.

Reply:
I'll take credit for the concept but @jasonmalinowski gets all the credit for making a suggestion that kept it this clean.